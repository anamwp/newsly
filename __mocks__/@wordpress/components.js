const React = require('react');

module.exports = {
	TextControl: jest.fn(() =>
		React.createElement('div', { 'data-testid': 'text-control' })
	),
	ColorPicker: jest.fn(() =>
		React.createElement('div', { 'data-testid': 'color-picker' })
	),
	Panel: jest.fn(({ children }) =>
		React.createElement('div', { 'data-testid': 'panel' }, children)
	),
	PanelBody: jest.fn(({ children, title }) =>
		React.createElement(
			'div',
			{ 'data-testid': 'panel-body', 'data-title': title },
			children
		)
	),
	PanelRow: jest.fn(({ children }) =>
		React.createElement('div', { 'data-testid': 'panel-row' }, children)
	),
	FontSizePicker: jest.fn(() =>
		React.createElement('div', { 'data-testid': 'font-size-picker' })
	),
	CheckboxControl: jest.fn(() =>
		React.createElement('div', { 'data-testid': 'checkbox-control' })
	),
};
