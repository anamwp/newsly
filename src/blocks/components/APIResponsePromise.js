/**
 * Manage API response promise
 * @param {string} url - URL of the API
 * @returns
 */
const APIResponsePromise = async (url = '') => {
	const options = {
		method: 'GET',
		headers: {
			accept: 'application/json',
			Authorization:
				'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI5NDQxMzQ5MmRiNWUyZTRjYTVlOTM0MDJjYTYyM2ZjYSIsIm5iZiI6MTcxOTIwNzU0OC45NzY5OCwic3ViIjoiNjY3OTA0YWNlZmRiOGMxNzc0MGI1MmZkIiwic2NvcGVzIjpbImFwaV9yZWFkIl0sInZlcnNpb24iOjF9.3LtMcOLpN8GfR8UiFDFPUYYHJVft69TrEzPssuTqnBA',
		},
	};
	const getMovieAPIResponse = await fetch(url, options);
	const getMovieAPIResponseJSON = await getMovieAPIResponse.json();
	if (getMovieAPIResponseJSON.success === false) {
		throw new Error(getMovieAPIResponseJSON.status_message);
	}
	return getMovieAPIResponseJSON;
};

export default APIResponsePromise;
