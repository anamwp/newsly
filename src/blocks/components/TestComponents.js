import React from 'react';

export function TestComponents() {
	return (
		<div
			style={{
				color: 'red',
				backgroundColor: 'yellow',
				fontWeight: 'bold',
				fontSize: '20px',
			}}
		>
			TestComponents is updated again updated and again
		</div>
	);
}

export function TestButtonClick() {
	return (
		<button
			style={{
				color: 'white',
				backgroundColor: 'blue',
				fontWeight: 'bold',
				fontSize: '16px',
			}}
			onClick={() => console.log('Button clicked!')}
		>
			TestButtonClick
		</button>
	);
}

export function Sub(a, b) {
	return a - b;
}
