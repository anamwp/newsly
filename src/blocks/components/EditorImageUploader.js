import React from 'react';
import { __ } from '@wordpress/i18n';
import { RawHTML, useState, useRef, useEffect } from '@wordpress/element';
import {
	MediaUploadCheck,
	MediaUpload,
	MediaPlaceholder,
	BlockIcon,
} from '@wordpress/block-editor';

/**
 * Handle image upload
 * @param {*} param0
 * @returns
 */
export default function EditorImageUploader({
	attributes,
	setAttributes,
	componentTitle = 'Upload Image',
	imageIdAttribute,
	imageUrlAttribute,
	editorPreviewUrlAttribute,
}) {
	/**
	 * Preview image after image upload
	 */
	const mediaPreviewHandle = attributes[editorPreviewUrlAttribute] && (
		<img
			class="inline-block w-full shadow-md hover:shadow-lg transition-all rounded"
			src={attributes[editorPreviewUrlAttribute]}
		/>
	);
	/**
	 * Remove Image
	 */
	const removeImage = () => {
		setAttributes({
			[imageIdAttribute]: null,
			[imageUrlAttribute]: '',
			[editorPreviewUrlAttribute]: '',
		});
	};
	return (
		/**
		 * Resource link
		 * https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/media-upload/README.md
		 * https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/media-placeholder/README.md
		 */
		<MediaUploadCheck>
			<MediaUpload
				render={() => (
					<div class="relative">
						{attributes[imageIdAttribute] && (
							<div className="remove-image-button absolute right-4 top-5 z-10 ">
								<button
									class="components-button is-primary"
									onClick={removeImage}
								>
									Remove
								</button>
							</div>
						)}
						<MediaPlaceholder
							onSelect={(el) => {
								setAttributes({
									[imageUrlAttribute]: el.url,
									[editorPreviewUrlAttribute]:
										el.sizes.medium.url,
									[imageIdAttribute]: el.id,
								});
							}}
							multiple={false}
							accept="image/*"
							icon={<BlockIcon icon="format-image" />}
							allowedTypes={['image']}
							labels={{ title: componentTitle }}
							mediaPreview={mediaPreviewHandle}
						/>
					</div>
				)}
			/>
		</MediaUploadCheck>
	);
}
