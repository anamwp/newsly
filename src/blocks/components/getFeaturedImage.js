import React from 'react';
import { useSelect } from '@wordpress/data';
import { useEffect } from '@wordpress/element';

/**
 * ❗️ Before use this component make sure below data is available ❗️
 * 👉 - Block is having [selectedPostFeaturedImage] attr.
 * 👉 - function [SelectedPostFeaturedImage] is defined in parent component.
 * 
 * Sample function that can be passed as a prop to GetFeaturedImage
 * to handle the selected post featured image.
 * This function can be defined in the parent component.
	const SelectedPostFeaturedImage = (data) => {
		var featuredImageData = data;
		setAttributes({
			selectedPostFeaturedImage: featuredImageData,
		});
	};
 * 👉 this is how you can use this component
	<GetFeaturedImage
		postFeaturedMediaId={postData.featured_media}
		selectedPostFeaturedImage={
			SelectedPostFeaturedImage
		}
	/>
	👉 In the block JSON file attr needs to add like below
	"selectedPostFeaturedImage": {
		"type": "object",
		"default": {}
	}
 */

export default function GetFeaturedImage(props) {
	const { postFeaturedMediaId, selectedPostFeaturedImage } = props;

	const getMediaContent = useSelect(
		(select) => {
			return select('core').getEntityRecords('postType', 'attachment', {
				include: [postFeaturedMediaId],
			});
		},
		[postFeaturedMediaId]
	);
	/**
	 * Store fetch image content to block attribute
	 */
	useEffect(() => {
		if (getMediaContent && getMediaContent.length > 0) {
			selectedPostFeaturedImage(getMediaContent[0]); // Pass featured image to parent
		}
	}, [getMediaContent]);

	return (
		<div className="mb-3">
			{getMediaContent
				? getMediaContent.map((data) => {
						return (
							<img
								className="inline-block w-full"
								src={data.source_url}
								alt=""
							/>
						);
				  })
				: 'Fetching Image'}
		</div>
	);
}
