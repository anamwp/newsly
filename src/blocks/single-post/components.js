import React from 'react';
import apiFetch from '@wordpress/api-fetch';
import { RawHTML, useState, useRef, useEffect } from '@wordpress/element';

export default function RenderPostCategoryData(props) {
	let postArr = props.catArr.toString();
	const [catData, setCatData] = useState([]);
	const { attributes, setAttributes } = props.parentProps;

	useEffect(() => {
		if (!postArr) return;

		apiFetch({
			path: `/wp/v2/categories?include=${postArr}`,
		})
			.then((res) => {
				setCatData(res);
				if (
					JSON.stringify(res) !==
					JSON.stringify(
						props.parentProps.attributes.selectedPostCategory
					)
				) {
					props.updateCatAttrCallback(res);
				}
			})
			.catch((err) => console.log(err));
	}, []);

	return (
		<div>
			{!catData && <p>Fetching Data</p>}
			{catData &&
				catData.map((singleCat) => {
					return (
						<a
							href={singleCat.link}
							style={{ marginRight: '10px' }}
							className="inline-block no-underline text-xs text-slate-600 hover:text-slate-800 bg-slate-100 hover:bg-slate-300 capitalize p-1 mr-1 rounded-md transition-all"
						>
							{singleCat.name}
						</a>
					);
				})}
		</div>
	);
}
