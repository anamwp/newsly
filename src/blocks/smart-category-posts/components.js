import React, { useState, useEffect } from 'react';
import apiFetch from '@wordpress/api-fetch';
import { RawHTML, useRef } from '@wordpress/element';

export default function RenderPostCategoryData(props) {
	let postArr = props.catArr.toString();
	const [catData, setCatData] = useState(null);
	// const {attributes, setAttributes} = props.parentProps;

	useEffect(() => {
		if (postArr) {
			apiFetch({
				path: `/wp/v2/categories?include=${postArr}`,
			})
				.then((res) => {
					setCatData(res);
				})
				.catch((err) => console.log(err));
		}
	}, [postArr]);

	return (
		<div>
			{!catData && <p>Fetching Data</p>}
			{catData &&
				Array.isArray(catData) &&
				catData.map((singleCat, index) => {
					return (
						<a
							key={singleCat.id || index}
							href={singleCat.link}
							style={{ marginRight: '10px' }}
						>
							{singleCat.name}
						</a>
					);
				})}
		</div>
	);
}
