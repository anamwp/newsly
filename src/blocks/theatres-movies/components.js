import React from 'react';
import apiFetch from '@wordpress/api-fetch';
import { RawHTML, useState, useRef, useEffect } from '@wordpress/element';

export default function RenderPostCategoryData(props) {
	let postArr = props.catArr.toString();
	const [catData, setCatData] = useState();
	// const {attributes, setAttributes} = props.parentProps;

	useEffect(() => {
		apiFetch({
			path: `/wp/v2/categories?include=${postArr}`,
		})
			.then((res) => {
				setCatData(res);
			})
			.catch((err) => console.log(err));
	}, []);

	return (
		<div>
			{!catData && <p>Fetching Data</p>}
			{catData &&
				catData.map((singleCat) => {
					return (
						<a
							href={singleCat.link}
							style={{ marginRight: '10px' }}
						>
							{singleCat.name}
						</a>
					);
				})}
		</div>
	);
}
