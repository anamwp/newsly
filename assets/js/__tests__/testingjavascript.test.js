import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { TestComponents } from '../../../src/blocks/components/TestComponents';

describe('TestComponents', () => {
	it('renders correctly', () => {
		const { container } = render(<TestComponents />);
		expect(container).toMatchSnapshot();
	});
});
