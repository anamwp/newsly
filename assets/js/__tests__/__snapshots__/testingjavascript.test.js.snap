// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponents renders correctly 1`] = `
<div>
  <div>
    TestComponents
  </div>
</div>
`;
